airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/source/relationaldb/AbstractDbSource.java

```diff
@@ -399,15 +403,14 @@ private AutoCloseableIterator<AirbyteMessage> createReadIterator(final Database
       final JsonSchemaPrimitive cursorType = IncrementalUtils.getCursorType(airbyteStream,
           cursorField);
 
+      CursorStateMessageProducer messageProducer = new CursorStateMessageProducer(
+          stateManager,
+          cursorInfo.map(CursorInfo::getCursor).orElse(null));
```

Comment:
Wouldn't we be better off passing an `Optional` type here? We should avoid passing `null`s.

Reply:
fixed